Library/Homebrew/cmd/commands.rb

```diff
@@ -27,18 +27,30 @@ def commands
   end
 
   def internal_commands
-    (HOMEBREW_LIBRARY_PATH/"cmd").children.select(&:file?).map { |f| f.basename(".rb").to_s }
+    find_internal_commands HOMEBREW_LIBRARY_PATH/"cmd"
   end
 
   def internal_development_commands
-    (HOMEBREW_LIBRARY_PATH/"dev-cmd").children.select(&:file?).map { |f| f.basename(".rb").to_s }
+    find_internal_commands HOMEBREW_LIBRARY_PATH/"dev-cmd"
   end
 
   def external_commands
-    paths.flat_map { |p| Dir["#{p}/brew-*"] }.
-      select { |f| File.executable?(f) }.
-      map { |f| File.basename(f, ".rb")[5..-1] }.
-      reject { |f| f =~ /\./ }.
-      sort
+    paths.reduce([]) do |cmds, path|
+      Dir["#{path}/brew-*"].each do |file|
+        next unless File.executable?(file)
+        cmd = File.basename(file, ".rb")[5..-1]
+        cmds << cmd unless cmd.include?(".")
+      end
+      cmds
+    end.sort
+  end
+
+  private
+
+  def find_internal_commands(directory)
+    directory.children.reduce([]) do |cmds, f|
+      cmds << f.basename(".rb").to_s.sub(/\.sh$/, "") if f.file?
```

Comment:
> Previously, it would have ignored everything with an unsupported file extension.

I don’t think so. The previous code was:
```rb
(HOMEBREW_LIBRARY_PATH/"dev-cmd").children.select(&:file?).map { |f| f.basename(".rb").to_s }
```

I originally had more complicated code to avod this but I simplified it for this PR. If you edit command files and run `brew commands` at the same time you might end up with commands like `foo.rb~` and/or `.foo.rb.swp` if you use Vim. We could add a filter like `/^-*[a-z-]+$/` but this issue is unlikely to arise if you don’t touch the code so I’m not sure it’s necessary.